### DISCUSSION TITLE
.pth files always contribute to slower startup

### DISCUSSION THREAD

**Discussion by ericsnowcurrently** [faster-cpython/ideas#211] [General]
See #207 for one example.  In that case there's more than just the cost of reading the .pth file.  However, in general, every .pth file adds a cost to startup.

Any thoughts on what to do about it?

**Comment by markshannon** [comment]
And?
Do you have a plan to mitigate the cost?

**Reply by ericsnowcurrently** [reply]
I do not.  My intent was to have a place to discuss the situation.  I'll update the post.

**Reply by warsaw** [reply]
There have been lots of discussions about trying to do Something with pth files.  One problem is that they can be abused to run arbitrary Python code, whereas their original intent was to extend `sys.path`.  I'm in favor of migrating away from allowing arbitrary code to run from a pth file.

I'm personally not very happy about pth files being used to extend `sys.path`, because they are exceedingly difficult to debug when things go wrong.  That's probably a controversial stance because people rely on this path extending feature of pth files.

**Reply by ericsnowcurrently** [reply]
FWIW, I stand with you on this.
